src/remote.c

```diff
@@ -670,18 +670,26 @@ int git_remote_set_pushurl(git_repository *repo, const char *remote, const char*
 	return set_url(repo, remote, CONFIG_PUSHURL_FMT, url);
 }
 
-const char* git_remote__urlfordirection(git_remote *remote, int direction)
+static const char *git_remote__urlresolve(const char *url, int direction, const git_remote_callbacks *callbacks)
+{
+    if (callbacks != NULL && callbacks->resolve_url != NULL) {
```

Comment:
`if (callbacks && callbacks->resolve_url)`

Same as above, this should return an integer error code so that we may distinguish fatal errors like e.g. OOM situations